pine/src/main/java/com/distkv/pine/components/cache/PineCacheImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.distkv.pine.components.cache;
+
+import com.distkv.client.DistkvClient;
+import com.distkv.common.exception.KeyNotFoundException;
+import com.distkv.pine.components.AbstractPineHandle;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class PineCacheImpl extends AbstractPineHandle implements PineCache {
+
+  private static final String COMPONENT_TYPE = "CACHE";
+
+  private DistkvClient distkvClient;
+
+  private Long expireTime;
+
+  final Set<String> set = new HashSet<>();
+
+  public PineCacheImpl(DistkvClient distkvClient, Long expireTime) {
+    super();
+    this.distkvClient = distkvClient;
+    this.expireTime = expireTime;
+  }
+
+  protected String getComponentType() {
+    return COMPONENT_TYPE;
+  }
+
+  @Override
+  public void newItem(String item) {
+    try {
+      distkvClient.strs().get(item);
+      distkvClient.expire(item, expireTime);
+      set.add(item);
+    } catch (KeyNotFoundException | InvalidProtocolBufferException e) {
+      distkvClient.strs().put(item, item);
+      distkvClient.expire(item, expireTime);
+      set.add(item);
```

Comment:
Why put a chae into Set? Why store an item repeatedly local service？

Reply:
There may be not Suitable。 I will replace it soon。 i think set will filled all the keys which include expired or not expired 。so it distinct key is keyNotFound or isExpired.  i think key has 
three status 1.expired 2.not expired.3 never appeared .a total new one